### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by Nasirkhan294**
### In this milestone I:

- [ ] Added a classroom class for student
- [ ] Added a rental class
- [ ] Added a book class

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+  end
+
+  def add_rental(person, date)
+    rental = Rental.new(date, self, person)
+    @rentals.push(rental)
```
Comment (Olamarx) [inline book.rb]: - [ ] You need to add the rentals to the attr_accessor for the getter and setter too. Inclusive, you need to initialize the array in the initialize method too.

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :date
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
```
Comment (Olamarx) [inline rental.rb:7]: - [ ] Getters and Setters for @book and @person instance variables should also be present.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @Nasirkhan294, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- All methods in place ✔️
- Readme file professional ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb [hunk 3]
```diff
@@ -1,9 +1,17 @@
...
+
   def initialize(classroom, age, name = 'Unknown', parent_permission: true)
     super(age, name, parent_permission)
     @classroom = classroom
+    add_classroom(classroom)
+  end
+
+  def add_classroom(classroom)
+    @classroom = classroom
+    classroom.students.push(self) unless classroom.students.include?(self)
```
Comment (mckent05) [inline student.rb:14]: **OPTIONAL** - Good job attempting to create a custom setter for the Student class, but you have not implemented it properly. I suggest you do this:
```ruby
def add_classroom=(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self)
end
```

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(person, date)
+    rental = Rental.new(date, self, person)
+    @rentals.push(rental)
```
Comment (mckent05) [inline book.rb]: **OPTIONAL** - Good job done so far, but I suggest that you do not need to push the created rental into the rentals attribute. That was already done when you created a new rental instance in the code directly above. Please fix this 🙏

**Review by mckent05 — APPROVED** [review]
Hi @Nasirkhan294, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- Good job setting up the associations between your classes 👍 👍 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**